src/components/Router.js

```diff
@@ -20,12 +27,27 @@ class Router extends Component {
     }
   }
 
+  getAuthenticatedUser() {
+    Auth.currentAuthenticatedUser()
+      .then(user => {
+        const email = user.attributes.email
+        if (email.substring(email.indexOf("@") + 1, email.length) === 'ubcbiztech.com') {
+          this.props.setUser(user)
+        }
+        else {
+          Auth.signOut()
+          alert('You must use a ubcbiztech.com email')
+        }
+      })
+      .catch(() => console.log("Not signed in"))
+  }
+
   componentDidMount() {
     fetch(API_URL + "/events/get", {
     })
       .then((response) => response.json())
       .then((response) => {
-        this.setState({
+        this.props.setEvents({
```

Comment:
Out of curiosity, what is the reason for switching from setting the state to props? Is it because props are immutable?

Reply:
props are connected to the redux store
So now the events are sent to the redux store instead of this component's state

Reply:
Sorry, I didn't read your commit message

Reply:
Ahh, and this is because we're using the library router instead of your own?

Reply:
Not necessarily, its just cleaner to have it in the store instead of passing it down the component tree (we could have done with with my router as well)